x-pack/plugins/ml/public/application/util/custom_url_utils.ts

```diff
@@ -18,11 +19,33 @@ import {
   CustomUrlAnomalyRecordDoc,
 } from '../../../common/types/custom_urls';
 import { AnomalyRecordDoc } from '../../../common/types/anomalies';
+import type { DataGridItem } from '../components/data_grid';
 
 // Value of custom_url time_range property indicating drilldown time range is calculated automatically
 // depending on the context in which the URL is being opened.
 const TIME_RANGE_AUTO = 'auto';
 
+// Replaces the $ delimited tokens in the url_value of the custom URL configuration
+// with values from the supplied document.
+export function replaceTokensInDFAUrlValue(
+  customUrlConfig: UrlConfig | KibanaUrlConfig,
+  doc: DataGridItem,
+  timeRange?: TimeRange
+) {
+  // If urlValue contains $earliest$ and $latest$ tokens, add in times to the test doc.
+  const urlValue = customUrlConfig.url_value;
+  const record = { ...doc } as any;
```

Comment:
can we get rid of any?

Reply:
Removed in 642ce6dbe70bb719b6416e5a61184a6b966b71d7